src/app/modules/workspace/workspace.utils.ts

```diff
@@ -0,0 +1,33 @@
+import { StatusCodes } from 'http-status-codes'
+
+import { createLoggerWithLabel } from '../../config/logger'
+import { ApplicationError, DatabaseError } from '../core/core.errors'
+import { ErrorResponseData } from '../core/core.types'
+
+const logger = createLoggerWithLabel(module)
+
+export const mapRouteError = (
+  error: ApplicationError,
+  coreErrorMessage?: string,
+): ErrorResponseData => {
+  switch (error.constructor) {
+    case DatabaseError:
```

Comment:
what about the other specific error classes returned by `transformMongoError`, like `DatabaseValidationError` or `DatabaseConflictError`? should those return different status codes?

Reply:
Will update to include the other MongoErrors. Yes I think those should return different status codes